### PR TITLE
MTM-51435 rename classes, create client builder

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: java-generic-mqtt/src/main/java/com/cumulocity/generic/mqtt/client/GenericMqttClient.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package com.cumulocity.generic.mqtt.client;
+
+import com.cumulocity.generic.mqtt.client.websocket.GenericMqttClientBuilder;
+
+/**
+ * {@link GenericMqttClient} is used to configure and create instances of {@link GenericMqttPublisher} and {@link GenericMqttSubscriber}.
+ */
+public interface GenericMqttClient {
```
Comment (rohitjoshi-c8y) [inline java-generic-mqtt/src/main/java/com/cumulocity/generic/mqtt/client/GenericMqttClient.java]: 👍

File: java-generic-mqtt/src/main/java/com/cumulocity/generic/mqtt/client/GenericMqttConfig.java [hunk 2]
```diff
@@ -5,12 +5,12 @@
 
 @Builder
 @Getter
-public class GenericMqttConnectionConfig {
```
Comment (rohitjoshi-c8y) [inline java-generic-mqtt/src/main/java/com/cumulocity/generic/mqtt/client/GenericMqttConfig.java]: its specific to connection so I am not sure how best we communicate that

File: java-generic-mqtt/src/main/java/com/cumulocity/generic/mqtt/client/GenericMqttListener.java [hunk 3]
```diff
@@ -6,9 +6,9 @@
 import java.net.URI;
 
 /**
- * Implement this interface to handle events from the websocket server after {@link GenericMqttSubscriber#subscribe(GenericMqttMessageListener)}.
+ * Implement this interface to handle events from the websocket server after {@link GenericMqttSubscriber#subscribe(GenericMqttListener)}.
  */
-public interface GenericMqttMessageListener {
+public interface GenericMqttListener {
```
Comment (rohitjoshi-c8y) [inline java-generic-mqtt/src/main/java/com/cumulocity/generic/mqtt/client/GenericMqttListener.java]: I think we need to convey its MESSAGE listener I think

Comment (ghost) [reply java-generic-mqtt/src/main/java/com/cumulocity/generic/mqtt/client/GenericMqttListener.java]: hmm to so maybe we should remove onOpen/onClose methods and then we can call it MessageListener? as in current shape it is not only message listener :)

Comment (ghost) [reply java-generic-mqtt/src/main/java/com/cumulocity/generic/mqtt/client/GenericMqttListener.java]: javadoc should not mention about web socket implementation anyway :) and both handlers won't be available if we switch to pulsar client. What do you think?

File: java-generic-mqtt/src/main/java/com/cumulocity/generic/mqtt/client/websocket/GenericMqttWebSocketClient.java [hunk 4]
```diff
@@ -1,26 +1,58 @@
...
+            @Override
+            public GenericMqttPublisherBuilder topic(final String topic) {
+                config.setTopic(topic);
+                return this;
+            }
+
+            @Override
+            public GenericMqttPublisherBuilder connectionTimeout(final long timeoutInMillis) {
+                config.setConnectionTimeout(timeoutInMillis);
+                return null;
```
Comment (rohitjoshi-c8y) [inline java-generic-mqtt/src/main/java/com/cumulocity/generic/mqtt/client/websocket/GenericMqttWebSocketClient.java]: return null?

**Pull request merged**